### ISSUE TITLE
False positive used-before-assignment with inline if

### ISSUE THREAD

**Issue by Zeckie** [PyCQA/pylint#7779]
### Bug description

```python
str(foo if (foo := "BAR") else "").lower()
```

Pylint gives error: `foobar.py:1:4: E0601: Using variable 'foo' before assignment (used-before-assignment)`

When run in python:
```python
>>> str(foo if (foo := "BAR") else "").lower()
'bar'
```

### Command used

```shell
pylint foobar.py
```

### Pylint output

```shell
************* Module foobar
foobar.py:1:4: E0601: Using variable 'foo' before assignment (used-before-assignment)
```

### Expected behavior

No errors

### Pylint version

```shell
pylint 2.15.5
astroid 2.12.12
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
```

### OS / Environment

Windows

**Comment by nickdrozd** [comment]
Variations that don't raise the error:

```python
# reverse if / else values
str("" if (foo := "BAR") else foo).lower()

# drop .lower()
str(foo if (foo := "BAR") else "")

# simple assignment
var = foo if (foo := "BAR") else ""
```
